package excecao;

public class Causa {

	public static void main(String[] args) {
		Aluno a1 = new Aluno("Gui", 8);
		metodoA(a1);
		metodoA(null);

	}
	
	static void metodoA(Aluno aluno) {
		try {
			metodoB(aluno);
		} catch (Exception causa) {
			throw new IllegalArgumentException(causa);
		}
	}

	static void metodoB(Aluno aluno) {
		if(aluno == null) {
			throw new NullPointerException("Aluno est√° nulo");
		}
		System.out.println(aluno.nome);
	}

}
